src/plugins/expression_reveal_image/public/components/reveal_image_component.tsx

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useRef, useState, useEffect, useCallback } from 'react';
+import { IInterpreterRenderHandlers } from 'src/plugins/expressions';
+import { NodeDimensions, RevealImageRendererConfig, OriginString } from '../../common/types';
+import { isValidUrl, elasticOutline } from '../../../presentation_util/public';
+import './reveal_image.scss';
+
+interface RevealImageComponentProps extends RevealImageRendererConfig {
+  handlers: IInterpreterRenderHandlers;
+  parentNode: HTMLElement;
+}
+
+interface ImageStyles {
+  width?: string;
+  height?: string;
+  clipPath?: string;
+}
+
+interface AlignerStyles {
+  backgroundImage?: string;
+}
+
+function RevealImageComponent({
+  handlers,
+  parentNode,
+  percent,
+  origin,
+  image,
+  emptyImage,
+}: RevealImageComponentProps) {
+  const [loaded, setLoaded] = useState(false);
+  const [dimensions, setDimensions] = useState<NodeDimensions>({
+    width: 1,
+    height: 1,
+  });
+
+  const imgRef = useRef<HTMLImageElement>(null);
+
+  // modify the top-level container class
+  parentNode.className = 'revealImage';
+
+  // set up the overlay image
+  const updateImageView = useCallback(() => {
+    if (imgRef.current) {
+      setDimensions({
+        height: imgRef.current.naturalHeight,
+        width: imgRef.current.naturalWidth,
+      });
+
+      setLoaded(true);
+      handlers.done();
+    }
+  }, [imgRef, handlers]);
+
+  useEffect(() => {
+    handlers.event({ name: 'onResize', data: updateImageView });
+    handlers.done();
+    return () => {
+      handlers.event({ name: 'destroy' });
```

Comment:
where is this event consumed ?

Reply:
> where is this event consumed ?

src/plugins/expression_reveal_image/public/expression_renderers/reveal_image_renderer.tsx:32

Reply:
And you can find such a workflow in every `renderer`.

Reply:
that's not consuming this event, onDestroy renderer is a separate thing. This event will get fired and the consumer (canvas app) would need to listen for it and react to it, renderers just won't get any information about this.

can you explain the workflow ?

Reply:
You were right. The workflow is further: 
1. For example, a user is pressing `backspace` after selecting the element.   
`x-pack/plugins/canvas/public/components/workpad_page/workpad_interactive_page/interactive_workpad_page.js` is rendering `WorkpadShortcuts`. 
Among them: `x-pack/plugins/canvas/public/lib/element_handler_creators.ts:63`. On `backspace` `deleteNodes` is executed. It is removing elements from `redux state` and calling rerender.
2. After current rerender, `componendWillUnmount` is executed. We are writing in a functional way with `hooks`, so the returned function from `useEffect` is called at `RenderWithFn` component. (`x-pack/plugins/canvas/public/components/render_with_fn/render_with_fn.tsx:84`). It is emitting `destroy` on line 84.
3. Renderer is listening to current `event`, by binding `handlers.onDestroy(...)` at `src/plugins/expression_reveal_image/public/expression_renderers/reveal_image_renderer.tsx:31`. On `destroy` it is unmounting component, which has been mounted in parent node.
4. Component, which has been rendered by renderer is unmounting and calling `event({ name: 'destroy' })`. Before I believe, there was some idea behind it, but now it is useless. 
Thank you, @ppisljar,  for this suggestion.